### PR TITLE
UGENE-6988. Replace GUITestIgnorable class functionality with test la…

### PR DESCRIPTION
**Pull request by yalgaer**
…bels.

### CODE REVIEW

File: src/corelibs/U2Test/src/gui_tests/UGUITest.cpp [hunk 1]
```diff
@@ -131,12 +132,39 @@ static QString getScreenshotDir() {
     return result;
 }
 
+/** Returns true if the test is has Ignored or IgnoredOn<CurrentOS>. */
```
Comment (atiunov) [inline src/corelibs/U2Test/src/gui_tests/UGUITest.cpp]: "is has"

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 2]
```diff
@@ -116,62 +132,49 @@ static int minutes(int minutes) {
...
-        auto test = new TestClass(DEFAULT_GUI_TEST_TIMEOUT, labels(Nightly)); \
-        test->setIgnored(BY, MESSAGE); \
-        test->setReason(reason); \
-        guiTestBase->registerTest(test); \
-    }
+    guiTestBase->registerTest(new TestClass(TIMEOUT, nightly()));
 
-/** Registers a GUI test included into nightly build with a custom timeout and sets its state to Ignored. */
-#define REGISTER_TEST_IGNORED_BY_WITH_TIMEOUT(TestClass, TIMEOUT, BY, MESSAGE, reason) \
+/** Registers a GUI with a given set of labels and description. Usually used to pass some non-default labels and the reason to the test. */
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: "Registers a GUI ->test<- with a given"

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 3]
```diff
@@ -116,62 +132,49 @@ static int minutes(int minutes) {
...
 /** Registers a GUI test with the labels provided.*/
 #define REGISTER_TEST_L(TestClass, LABELS) REGISTER_TEST_TL(TestClass, DEFAULT_GUI_TEST_TIMEOUT, LABELS);
 
-namespace U2 {
+/**  Registers test on all platforms but with a global Ignored label. */
+#define REGISTER_TEST_IGNORED(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(Ignored), MESSAGE)
+/**  Registers test on all platforms but with an IgnoredOnWindows label. */
+#define REGISTER_TEST_IGNORED_WINDOWS(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(IgnoredOnWindows), MESSAGE)
+/**  Registers test on all platforms but with an IgnoredOnMacOS label. */
+#define REGISTER_TEST_IGNORED_MAC(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(IgnoredOnMacOS), MESSAGE)
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: REGISTER_TEST_IGNORED_LINUX should be added.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 4]
```diff
@@ -116,62 +132,49 @@ static int minutes(int minutes) {
...
+
+/** Registers test on all platforms except Windows. */
+#define REGISTER_TEST_NOT_FOR_WINDOWS(TestClass) REGISTER_TEST_L(TestClass, labels(Nightly, Linux, MacOS))
+/** Registers test on all platforms except MacOS. */
+#define REGISTER_TEST_NOT_FOR_MAC(TestClass) REGISTER_TEST_L(TestClass, labels(Nightly, Linux, Windows))
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: REGISTER_TEST_NOT_FOR_LINUX should be added.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 5]
```diff
@@ -116,62 +132,49 @@ static int minutes(int minutes) {
...
+
+/** Registers test on all platforms and adds ignored labels for all of them, but not for the Linux. */
+#define REGISTER_TEST_LINUX(TestClass, MESSAGE) REGISTER_TEST_EXT(TestClass, nightly(IgnoredOnMacOS, IgnoredOnWindows), MESSAGE)
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: This define looks very strange. It does the same as REGISTER_TEST_ONLY_LINUX does. I offer to use REGISTER_TEST_ONLY_LINUX, because it is much more clear.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 6]
```diff
@@ -105,8 +104,25 @@
 #include "tests/regression_scenarios/GTTestsRegressionScenarios_5001_6000.h"
 #include "tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.h"
 
-static QStringList labels(const QString &label) {
-    return QStringList() << label;
+namespace U2 {
+
+/** Converts list of label args into QStringList. */
+static QStringList labels(const QString &label1, const QString &label2 = "", const QString &label3 = "", const QString &label4 = "", const QString &label5 = "") {
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Hardcoded parameters will cause additional unnecessary changes if additional labels appear. I offer to receive a QStringList as parameter (or QList<QString>, if compiler will comply) and initialize the list where the method calls. For example: labels({Metagenomics, Linux})).

**Review by atiunov — CHANGES_REQUESTED** [review]

File: src/corelibs/U2Test/src/gui_tests/UGUITest.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Test/src/gui_tests/UGUITest.cpp]: Fixed.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 2 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Fixed.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 3 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: This macro will be unused.
The final goal is to remove all ignored macros, so I was happy to remove this one.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 5 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: I agree, but I kept the old naming.
I can give it a new name if you want but I think in this case it should have IGNORED word in the name.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 6 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Good suggestion. I was not aware of this syntax. Fixed.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 4 above)
Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: This macro will be unused today. I suggest adding it later when it will really be needed.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp [hunk 7]
```diff
@@ -107,20 +107,13 @@
...
-static QStringList labels(const QString &label1, const QString &label2 = "", const QString &label3 = "", const QString &label4 = "", const QString &label5 = "") {
-    QStringList rawLabelList = QStringList() << label1 << label2 << label3 << label4 << label5;
-    QStringList resultLabelList;
-    for (const QString &rawLabel : rawLabelList) {
-        if (!rawLabel.isEmpty()) {
-            resultLabelList << rawLabel;
-        }
-    }
-    return resultLabelList;
+static QStringList labels(const QStringList &labelList) {
```
Comment (atiunov) [inline src/plugins/GUITestBase/src/GUITestBasePlugin.cpp:110]: Do you really need this function now?

Comment (yalgaer) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp:110]: When I pass {label1, label2, label3} to macro I see a compiler error about the wrong number of params.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 3 above)
Comment (atiunov) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp]: Ok.

File: src/plugins/GUITestBase/src/GUITestBasePlugin.cpp (hunk 7 above)
Comment (atiunov) [reply src/plugins/GUITestBase/src/GUITestBasePlugin.cpp:110]: Ok. Add a comment.

**Review by atiunov — APPROVED** [review]
